### PR TITLE
docs: Update design info with more Storybook details

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Introduces storybook a bit more for new users and images the image.

I also squeezed in a metadata fix.

image

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4767-gjurpmferwwwb2cxosq6gociq8pq3um6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603f0ad765f817faadffbf21

File: docs/dls/contributing-to-storybook.md [hunk 1]
```diff
@@ -4,8 +4,38 @@ title: Contributing to Storybook
 description: Documentation on How to Contribute to Storybook
 ---
 
-You find our storybook at
-[http://backstage.io/storybook](http://backstage.io/storybook)
+Storybook is an open source tool to provide users a way to explore the Backstage
```
Comment (Rugvip) [inline docs/dls/contributing-to-storybook.md:7]: Feeling that this implies that storybook is built for for this purpose specifically, which isn't quite the case 😅

**Review by Rugvip — APPROVED** [review]
Thanks, looks good! 👍 

Just an optional nit if you agree

File: docs/dls/contributing-to-storybook.md (hunk 1 above)
Comment (adamdmharvey) [reply docs/dls/contributing-to-storybook.md:7]: Yep; great point. Will reword.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4767-gjurpmferwwwb2cxosq6gociq8pq3um6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603f0ad765f817faadffbf21

**Pull request merged**